packages/server/modules/blobstorage/services/management.ts

```diff
@@ -61,3 +68,71 @@ export const getFileStreamFactory =
     // eslint-disable-next-line @typescript-eslint/no-unsafe-return
     return await getObjectStream({ objectKey: objectKey! })
   }
+
+type UpdateBlobMetadataDeps = {
+  getBlobMetadata: GetBlobMetadata
+  updateBlob: UpdateBlob
+}
+
+const updateBlobMetadataFactory =
```

Comment:
All 3 markUploadXXX functions are small and very similar with the same set of dependencies